### DISCUSSION TITLE
show all typescript errors when build, instead of only first error

### DISCUSSION THREAD

**Discussion by chauchakching** [vercel/next.js#17300] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

I'm working on a migration from flow js to typescript, and trying to fix the type errors after the conversion. 
It's annoying that the type check during build only shows the first error, instead of all errors. 

https://github.com/zeit/next.js/blob/d20660d6769a4408001b4b89f6d384ce62c04231/packages/next/lib/typescript/runTypeCheck.ts#L42

I don't know if there are going to be 10 more type errors, or 1000 more, and see the full picture.

## Describe the solution you'd like

List out all typescript type errors during build.

**Comment by andrasferenczi** [comment]
Running `npx tsc --noEmit` in the root shows all errors.

**Reply by billytetrud** [reply]
I tried adding noEmit: true in my tsconfig.json, but that doesn't seem to display multiple errors.

**Reply by ao-stepan** [reply]
@billytetrud  the same issue for me, but just running `npx tsc --noEmit` helps me

**Reply by tiborgg** [reply]
Even if this works, a compiler which does not show you all the errors is not a proper compiler.

**Reply by jussihirvi** [reply]
It seems that tsc is not honoring the `tsconfig` file at all, unless explicitely directed to do so using the `--project` flag followed by a path to its configuration file, or to a folder with a `tsconfig.json`.

**Reply by Stan-Stani** [reply]
I've run into this sort of issue when using Next with Turbo Repo.

My solution is to manually go into the specific workspace subdir and do `npx tsc --noEmit && yarn next lint`.


*ETA* Which show more issues than actually exist because it doesn't necessarily respect tsconfigs in other packages in the workspace and can thus end up applying the wrong .tsconfgs. 

For complex workspaces, use lewandom's suggestion below.

**Comment by lewandom** [comment]
As crappy as it sounds, running with `__NEXT_TEST_MODE` env var set, e.g.:
```
__NEXT_TEST_MODE=1 yarn build
```

prints the entire set of compile errors in `[Test Mode]`. See https://github.com/vercel/next.js/blob/canary/packages/next/src/lib/typescript/runTypeCheck.ts#L99.

Not sure if there are any side aspects of running build with this flag.
